Doc: added triage section to new contributor docs (#28347)
matplotlib/matplotlib: doc/devel/contribute.rst

.. _`open an issue`: https://github.com/matplotlib/matplotlib/issues/new?assignees=&labels=Documentation&projects=&template=documentation.yml&title=%5BDoc%5D%3A+

.. _contribute_triage:

Triage
------
We appreciate help keeping our `issue tracker <https://github.com/matplotlib/matplotlib/issues>`_
organized because the issue tracker is our centralized location for feature requests,
bug reports, tracking major projects, and discussing priorities.

Some examples of what we mean by triage are:

* labeling issues and pull requests
* verifying bug reports
* debugging and resolving issues
* linking to related issues, discussion, and external work

> Helping out with triage is discussed in detail in :ref:`bug_triaging`.

Reviewer: With my non-native English, I feel "helping out" is not that welcoming. I read it as "We have a deficiency. Can you temporarily take over a task that we currently cannot fulfill".

I suggest either

```suggestion
Helping with triage is discussed in detail in :ref:`bug_triaging`.
```
or
```suggestion
These tasks are discussed in detail in :ref:`bug_triaging`.
```
But you know the language better, so please decide whether to keep as is or change.

.. _`open an issue`: https://github.com/matplotlib/matplotlib/issues/new?assignees=&labels=Documentation&projects=&template=documentation.yml&title=%5BDoc%5D%3A+

.. _contribute_triage:

Triage
------
The `issue tracker <https://github.com/matplotlib/matplotlib/issues>`_ is our
centralized location for feature requests, bug reports,

> tracking major projects and discussing priorities. It takes a lot of
> time and effort to keep oragnized, so curating our issue tracker is a valuable
> contribution. Some examples are:

Reviewer: We can leave out the "time and effort" part. First, it doesn't sound very attractive to embark on something "taking a lot of time and effort", even if you don't have to do everything. Second, it sounds worse than it is. While it is work, I don't have the impression it's that much, e.g. compared to PR reviewing.

```suggestion
tracking major projects and discussing priorities. Helping to keep it
organized is a valuable contribution. Some examples are:
```

Alternative wording:

```suggestion
tracking major projects and discussing priorities. Curating it is a valuable
contribution. Some examples are:
```

* labeling issues and pull requests
* verifying bug reports
* debugging and resolving issues
* linking to related issues, discussion, and external work

> Helping out with triage is discussed in detail in :ref:`bug_triaging`.

Reviewer: As with code and documentation we should link the Contributor incubator. Code / documentation says

> If you get stuck, please reach out on the [Contributor incubator](https://matplotlib.org/devdocs/devel/contribute.html#contributor-incubator)

Likely that's not exactly the best the wording for triage. Which process do we want exactly? Is it ok that people just start to comment on issues? Or should they first contact us through the incubator.

Reviewer 2: Maybe something like

> If you have any questions about the process, please reach out on...

If someone is stuck about how to triage a particular issue, it's maybe better they just leave it for someone else to pick up (at least, that's what I do...!)

I think it's fine if people just start commenting on issues.  We are not overrun with people wanting to do this so I think it's pretty easy to keep track of what anyone new is doing.  We do not require people to ask first when making other contributions like creating PRs.

Triage help is a valuable contribution because it takes a lot of time and
judgement to effectively curate and resolve issues; therefore we very much
appreciate thoughtful feedback to issue authors. Please do not verbatim post
the output of a generative AI tool (such as chat GPT) as a response to a bug
report as we expect that the issue author has either already tried that or has a

> reason for not doing so.

Reviewer: This is a way of working aspect of triaging. It belongs under the above linked `bug triaging` docs, not here. - My idea of "Contributing guide > Getting started" is that it should give a high-level idea of the kinds contribution and direct interested people to the relevant parts of the documentation. Specific information on how (not) to do things should be in those parts of the documentation e.g. `bug triaging`, not on the overview page.

Reviewer: My first impulse was: This belongs in https://matplotlib.org/devdocs/devel/triage.html. But considering the definition of triage actions on that site, replying with a solution is beyond triaging. Triaging only makes sure the issue is well laid out and sorted. It does not solve it (except for invalid or duplicate issues). Therefore, I don't see that people will respond with AI as part of triaging. To make life simple here, I claim it's out of scope for the PR and we can leave that section out.

You could add something like "Proposing solutions" to the https://matplotlib.org/devdocs/devel/triage.html. It should be made clear that
- it's optional (a triager may add it if they already exactly know how to solve this)
- proposals / recommendations should be well considered and well laid out. If you are not sure, better leave the response to somebody else. This also includes auto-generated AI responses are not acceptable.
